src/runtime_src/core/common/query_requests.h

```diff
@@ -1073,6 +1075,26 @@ struct error : request
   }
 };
 
+// Retrieve asynchronous xocl errors from xocl driver
+struct xocl_errors : request
+{
+  using result_type = std::vector<char>;
+  static const key_type key = key_type::xocl_errors;
+
+  virtual boost::any
+  get(const device*) const = 0;
+
+  // Parse sysfs line and from class get error code and timestamp
+  XRT_CORE_COMMON_EXPORT
+  static std::pair<uint64_t, uint64_t>
+  to_value(const std::vector<char>& buf, xrtErrorClass ecl);
+
+  // Parse sysfs raw data and get list of errors
+  XRT_CORE_COMMON_EXPORT
+  static std::vector<xclErrorLast>
+  to_errors(const std::vector<char>& buf);
```

Comment:
I don't see this function used?

Reply:
For future use. Like in xbutil to dump all errors